internal/plugincommon/discovery/processor.go

```diff
@@ -109,12 +111,72 @@ func (cdp *ContractDiscoveryProcessor) Outcome(
 		}
 	}
 
+	// nonce manager address consensus
+	var nonceManagerAddrs [][]byte
+	for _, ao := range aos {
+		nonceManagerAddrs = append(
+			nonceManagerAddrs,
+			ao.Observation.DestNonceManager,
+		)
+	}
+
+	nonceManagerConsensus, err := getConsensusNonceManager("nonceManager", nonceManagerAddrs, fChain[cdp.dest])
+	if err != nil {
+		return dt.Outcome{}, fmt.Errorf("unable to reach consensus on nonce manager: %w", err)
+	}
+
 	// call Sync to bind contracts.
 	contracts := make(map[string]map[cciptypes.ChainSelector][]byte)
 	contracts[consts.ContractNameOnRamp] = plugincommon.GetConsensusMap(cdp.lggr, "onramp", onrampAddrs, fChain)
+	contracts[consts.ContractNameNonceManager] = map[cciptypes.ChainSelector][]byte{
+		cdp.dest: nonceManagerConsensus,
+	}
```

Comment:
Use the generic get consensus function
```suggestion
	contracts[consts.ContractNameNonceManager] = plugincommon.GetConsensusMap(cdp.lggr, "nonce manager", nonceManagerAddrs, fChain[cdp.dest])	}
```